### PR TITLE
kernel: bump 5.4 to 5.4.109

### PR DESCRIPTION
**Pull request by graysky2**
Ran update_kernel.sh in a fresh clone without any existing toolchains.

Manually rebased:
```
  pending-5.4/611-netfilter_match_bypass_default_table.patch
```
Build system: x86_64
Build-tested: ipq806x/R7800

Signed-off-by: John Audia <graysky@archlinux.us>

### CODE REVIEW

**Comment by rockdrilla** [comment]
Build system: x86_64
Build-tested: ath79/generic (TP-Link Archer C7 v2)
Run-tested: ath79/generic (TP-Link Archer C7 v2)

**Comment by timocapa** [comment]
```
Build system: x86_64
Build-tested: ramips/mt7621/{Xiaomi R3G, Redmi 2100}, Lantiq/XWAY/Fritz!Box 7320/7330
Run-tested: ramips/mt7621/{Xiaomi R3G, Redmi 2100}, Lantiq/XWAY/Fritz!Box 7320/7330
```
LGTM

File: target/linux/generic/pending-5.4/611-netfilter_match_bypass_default_table.patch [hunk 1]
```diff
@@ -80,7 +80,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 -	WARN_ON(!(table->valid_hooks & (1 << hook)));
 -	local_bh_disable();
  	addend = xt_write_recseq_begin();
--	private = rcu_access_pointer(table->private);
+-	private = READ_ONCE(table->private); /* Address dependency. */
```
Comment (dengqf6) [inline target/linux/generic/pending-5.4/611-netfilter_match_bypass_default_table.patch:83]: Another appearance of `private = rcu_access_pointer(table->private);` is at line 51, should it be replaced too?

Comment (graysky2) [reply target/linux/generic/pending-5.4/611-netfilter_match_bypass_default_table.patch:83]: I do not know.  I emailed the patch's author requesting a review.

Comment (hauke) [reply target/linux/generic/pending-5.4/611-netfilter_match_bypass_default_table.patch:83]: In this update of kernel 5.10 also the other place is changed: 
https://git.openwrt.org/?p=openwrt/staging/ldir.git;a=commitdiff;h=9473df3c6eaeaf98bbcebe1d0ec7333561d7ef56

Comment (adschm) [reply target/linux/generic/pending-5.4/611-netfilter_match_bypass_default_table.patch:83]: The patch itself essentially seems to move code, so I'd say "yes", too.

Comment (adschm) [reply target/linux/generic/pending-5.4/611-netfilter_match_bypass_default_table.patch:83]: The relevant kernel patch is a revert. So I've looked up our relevant commit introducing the reverted kernel patch:
https://github.com/openwrt/openwrt/commit/9b1b89229f0efef43a738f91875e0fd9fcc42277

This touches both locations, so "revert" both now. I will take care and probably merge the PR afterwards.

**Comment by adschm** [comment]
Thanks, merged.

**Pull request closed without merging**